src/adapter/src/coord/sequencer/inner/peek.rs

```diff
@@ -75,14 +76,25 @@ impl Coordinator {
     ) {
         event!(Level::TRACE, plan = format!("{:?}", plan));
 
+        let explain_ctx = match ctx.session().vars().emit_plan_insights_notice() {
+            true => {
+                let broken = false;
```

Comment:
Note that this means that any regular tracing spans / events emitted while the optimizer is running will be suppressed while `emit_plan_insights_notice` is `true`.

This is because you will be hitting the `else` branch in `OptimizerTrace::new(...)` constructor, which creates the subscriber chain rooted at `tracing_subscriber::registry()`. If you want to maintain the tracing capabilities if notices are turned on I suggest to set `broken = true`. We had some issues in the past where `broken = true` and using the `let subscriber = DelegateSubscriber::default()` was causing memory leaks, but we have not been seeing this recently.

Reply:
Isn't `broken = false` what `EXPLAIN ... SELECT` uses by default? Was just trying to match that standard flow!

Reply:
It is, but it currently has the somewhat unpleasant behavior of hiding all `tracing::span!` and `tracing::event!` calls that happen when you call
```rust
let _dispatch_guard = explain_ctx.dispatch_guard();
```
in the spawned optimization tasks because the `&dispatch` that is temporary set with
```rust
tracing::dispatcher::set_default(&dispatch)
```
is not rooted at the `DelegateSubscriber::default()` when `broken` is `false`. We did this because at some point I think there was a bug in `tokio::tracing` that was causing memory leaks (which so far were only visible when you run `EXPLAIN` in a hot loop like in the SQLSmith tests).

So far getting limited tracing support here had limited impact because it only affected `EXPLAIN` paths. If you really wanted to see `tracing::event!` lines for an `EXPLAIN <explainee>` statement (as opposed to just running `<explainee>` directly), you could do
```sql
ALTER SYSTEM SET enable_explain_broken = true;
```
and then type
```sql
EXPLAIN BROKEN <explainee> ...
```

However, because `ExplainContext::PlanInsightsNotice` is now set whenever `emit_plan_insights_notice` is set, it means that this behavior will extend to all `SELECT` queries that are executed from the console.

So my suggestion is to:

1. Set `let broken = true;` here for now in order to retain visibility in the console.
2. I'll prepare a PR (or a commit in this PR) that introduces a feature flag that forces `OptimizerTrace::new(...)` to be called with `true` when set, set the feature flag default to `true` everywhere, and do some local tests. If there are no issues with the next release we can remove the flag and the conditional in `OptimizerTrace::new(...)` with the next release.